src/Core/Services/MultipleMutationInputValidator.cs

```diff
@@ -0,0 +1,575 @@
+            {
+                // For the example createbooks mutation written above, the list value for `items` is interpreted as a List<IValueNode>
+                // i.e. items is a list of ObjectValueNode(s).
+                listOfIValueNode.ForEach(iValueNode => ValidateGraphQLValueNode(
+                    schema: schema,
+                    entityName: entityName,
+                    context: context,
+                    parameters: iValueNode,
+                    runtimeConfig: runtimeConfig,
+                    columnsDerivedFromParentEntity: columnsDerivedFromParentEntity,
+                    columnsToBeDerivedFromEntity: columnsToBeDerivedFromEntity,
+                    nestingLevel: nestingLevel,
+                    parentEntityName: parentEntityName,
+                    sqlMetadataProviderFactory: sqlMetadataProviderFactory));
+            }
+            else if (parameters is ObjectValueNode objectValueNode)
+            {
+                // For the example createbook mutation written above, the node for publisher field is interpreted as an ObjectValueNode.
+                // Similarly the individual node (elements in the list) for the reviews, authors ListValueNode(s) are also interpreted as ObjectValueNode(s).
+                ValidateObjectFieldNodes(
+                    schemaObject: ExecutionHelper.InputObjectTypeFromIInputField(schema),
+                    entityName: entityName,
+                    context: context,
+                    objectFieldNodes: objectValueNode.Fields,
+                    runtimeConfig: runtimeConfig,
+                    columnsDerivedFromSourceEntity: columnsDerivedFromParentEntity,
+                    columnsToDerive: columnsToBeDerivedFromEntity,
+                    nestingLevel: nestingLevel + 1,
+                    parentEntityName: parentEntityName,
+                    sqlMetadataProviderFactory: sqlMetadataProviderFactory);
+            }
+            else if (parameters is ListValueNode listValueNode)
+            {
+                // For the example createbook mutation written above, the list values for reviews and authors fields are interpreted as ListValueNode.
+                // All the nodes in the ListValueNode are parsed one by one.
+                listValueNode.GetNodes().ToList().ForEach(objectValueNodeInListValueNode => ValidateGraphQLValueNode(
+                    schema: schema,
+                    entityName: entityName,
+                    context: context,
+                    parameters: objectValueNodeInListValueNode,
+                    runtimeConfig: runtimeConfig,
+                    columnsDerivedFromParentEntity: columnsDerivedFromParentEntity,
+                    columnsToBeDerivedFromEntity: columnsToBeDerivedFromEntity,
+                    nestingLevel: nestingLevel,
+                    parentEntityName: parentEntityName,
+                    sqlMetadataProviderFactory: sqlMetadataProviderFactory));
+            }
```

Comment:
Add an else case that throws an exception with the message along the lines of parameters not being expected to be of any other type